### PR TITLE
CPM-515: Add SetNumberValue as a UserIntent to use for UpsertProductCommand

### PR DESCRIPTION
**Pull request by cemorin**
**Description (for Contributor and Core Developer)**
Add SetNumberValue as a UserIntent to use for UpsertProductCommand

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+namespace Akeneo\Pim\Enrichment\Product\Api\Command\UserIntent;
+
+final class SetNumberValue implements ValueUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private ?string $localeCode,
+        private ?string $channelCode,
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php:21]: Can you put the channel first, then the locale plz? It's a mistake we have done in the SetTextValue

File: src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php:6]: Not the good header

File: src/Akeneo/Pim/Enrichment/Product/back/Test/Specification/Api/Command/UserIntent/SetNumberValueSpec.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Akeneo PIM Enterprise Edition.
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Test/Specification/Api/Command/UserIntent/SetNumberValueSpec.php:6]: same

File: src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+ */
+final class SetNumberValue implements ValueUserIntent
+{
+    public function __construct(
+        private string $attributeCode,
+        private ?string $channelCode,
+        private ?string $localeCode,
+        private int $value
+    )
+    {
```
Comment (nmarniesse) [inline src/Akeneo/Pim/Enrichment/Product/back/Api/Command/UserIntent/SetNumberValue.php:19]:
```suggestion
    ) {
```

**Review by amigaire — APPROVED** [review]

**Comment by nmarniesse** [comment]
unfreeze

**Pull request merged**